frontend/src/app/app.component.ts

```diff
@@ -14,17 +16,24 @@ export class AppComponent implements OnInit {
 
   constructor(
     @Inject(WindowService) private window: any,
-    private router: Router
+    private router: Router,
+    private metaService: VolontuloMetaService,
+    private activatedRoute: ActivatedRoute
   ) { }
 
   ngOnInit() {
     this.window.ga('create', environment.googleAnalyticsAppID, 'auto');
-    this.router.events.subscribe(event => {
-      if (event instanceof NavigationEnd) {
+    this.router.events.pipe(
+      filter((event: NavigationEnd) => event instanceof NavigationEnd),
+      tap(event => {
         this.window.ga('set', 'page', event.urlAfterRedirects);
         this.window.ga('send', 'pageview');
-      }
-    });
+      }),
+      map(() => this.activatedRoute),
```

Comment:
It looks like You're a little bit overfocused on putting all login inside rxJS pipe (and passing it to other rxJS operator).
We can easly do here:
```
switchMap(() => {
  const route = this.activatedRoute;
  while (route.firstChild) {
    route = route.firstChild;
  }
  return route.data;
})
```
I have the same observation when You're changing:
```
if (event instanceof NavigationEnd) { ... }
```
into
```
filter((event: NavigationEnd) => event instanceof NavigationEnd),
```

BUT nevertheless what do You think about using here (I know, experimental) [`ActivationEnd`](https://angular.io/api/router/ActivationEnd):
```
// need to be class as it's used in runtime check. should be used for all components in decorator instead of @Component
class MetadataComponent extends Component {
  updatePageMetaData: () => {}: void;
}

this.router.events.subscribe(event => {
  if (event instanceof ActivationEnd) {
        const component = event.snapshot.component;
        if (component instanceof MetadataComponent) {
          component.updatePageMetaData();
        }
  }
}
```